client/ayon_maya/plugins/publish/validate_animation_product_type_publish.py

```diff
@@ -0,0 +1,44 @@
+import ayon_maya.api.action
+from ayon_core.pipeline.publish import (
+    PublishValidationError,
+    ValidateContentsOrder,
+)
+from ayon_maya.api import plugin
+
+
+class ValidateAnimationProductTypePublish(plugin.MayaInstancePlugin):
+    """Validate at least a single product type is exported for the instance.
+    
+    Validate either fbx animation collector or collect animation output 
+    geometry(Alembic) enabled for publishing otherwise no products
+    would be generated for the instance - publishing nothing valid.
+    """
+
+    order = ValidateContentsOrder
+    families = ["animation"]
+    label = "Animation Product Type Publish"
+    actions = [ayon_maya.api.action.SelectInvalidAction]
+
+    @classmethod
+    def get_invalid(cls, instance):
+        invalid = []
+        publish_attributes = instance.data["publish_attributes"]
+        if "animation.fbx" in instance.data["families"]:
+            return invalid
+        elif publish_attributes["ExtractAnimation"]["active"]:
+            return invalid
+        else:
+            invalid.append(instance.name)
+
+        return invalid
+
+    def process(self, instance):
+        invalid = self.get_invalid(instance)
+        if invalid:
+            message = (
+                f"Invalid Animation Product Type in {invalid}\n"
+                "Users must turn on either 'Collect Fbx Animation'\n"
+                "or 'Collect Animation Output Geometry(Alembic)'\n"
```

Comment:
Pretty sure this toggle doesn't exist anymore and it's now "Extract Animation" as label, which may actually be confusing since you wouldn't know whether it'd be disabled/enabling the Alembic or not?

Reply:
I admit this is a bit confusing. I improved the message in the updated commit@moonyuet
[support usd check on the validation](https://github.com/ynput/ayon-maya/pull/68/commits/13c4df7b097da478010d20a954508a15e6b696aa) but not sure it does still look confusing.